plugins/blockvolume/blockprovider/gluster-loopback/glusterloopback.go

```diff
@@ -0,0 +1,260 @@
+package glusterloopback
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/exec"
+
+	"github.com/gluster/glusterd2/glusterd2/volume"
+	"github.com/gluster/glusterd2/plugins/blockvolume/blockprovider"
+	"github.com/gluster/glusterd2/plugins/blockvolume/utils"
+
+	log "github.com/sirupsen/logrus"
+	config "github.com/spf13/viper"
+	"k8s.io/kubernetes/pkg/util/mount"
+)
+
+const providerName = "gluster-loopback"
+
+var mounter = mount.New("")
+
+func init() {
+	log.WithField("name", providerName).Infof("Registering block provider")
+	blockprovider.RegisterBlockProvider(providerName, newGlusterLoopBlk)
+}
+
+// GlusterLoopBlk implements block Provider interface. It represents a gluster-block
+type GlusterLoopBlk struct {
+	mounts map[string]string
+}
+
+func newGlusterLoopBlk() (blockprovider.Provider, error) {
+	gb := &GlusterLoopBlk{}
+
+	gb.mounts = make(map[string]string)
+
+	return gb, nil
+}
+
+func mountHost(g *GlusterLoopBlk, hostVolume string) (string, error) {
+	hostDir := g.mounts[hostVolume]
+	if hostDir == "" {
+		hostDir = config.GetString("rundir") + "/blockvolume/" + hostVolume
+		notMnt, err := mounter.IsLikelyNotMountPoint(hostDir)
+		if err != nil {
+			if os.IsNotExist(err) {
+				err = os.MkdirAll(hostDir, os.ModeDir|os.ModePerm)
+				if err != nil {
+					return "", fmt.Errorf("failed to create mount point %+v", err)
+				}
+				notMnt = true
+			} else {
+				return "", fmt.Errorf("failed to mount block host volume %+v", err)
+			}
+		}
+
+		if notMnt {
+			err = volume.MountVolume(hostVolume, hostDir, "")
+			if err != nil {
+				return "", fmt.Errorf("failed to mount block host volume %+v", err)
+			}
+		}
+		g.mounts[hostVolume] = hostDir
+	}
+	return hostDir, nil
+}
+
+// CreateBlockVolume will create a gluster block volume with given name and size having `hostVolume` as hosting volume
+func (g *GlusterLoopBlk) CreateBlockVolume(name string, size uint64, hostVolume string, options ...blockprovider.BlockVolOption) (blockprovider.BlockVolume, error) {
+	blockVolOpts := &blockprovider.BlockVolumeOptions{}
+	blockVolOpts.ApplyOpts(options...)
+	logger := log.WithFields(log.Fields{
+		"block_name":           name,
+		"hostvol":              hostVolume,
+		"requested_block_size": size,
+	})
+
+	hostDir, err := mountHost(g, hostVolume)
+	if err != nil {
+		return nil, fmt.Errorf("failed to mount block hosting volume %+v", err)
+	}
+
+	blockFileName := hostDir + "/" + name
+	cmd := exec.Command("truncate", fmt.Sprintf("-s %d", size), blockFileName) //nolint: gosec
+	err = cmd.Run()
+	if err != nil {
+		return nil, fmt.Errorf("failed to truncate block file %s: %+v", blockFileName, err)
+	}
+
+	cmd = exec.Command("mkfs.xfs", "-f", blockFileName) //nolint: gosec
```

Comment:
Use `utils.ExecuteCommandRun` for including stderr info in error

Reply:
Done